File: app/src/main/java/com/benoitquenaudon/tvfoot/red/app/injection/module/PreferenceServiceModule.java
package com.benoitquenaudon.tvfoot.red.app.injection.module;

import android.app.Application;
import android.content.SharedPreferences;
import dagger.Module;
import dagger.Provides;

import static android.content.Context.MODE_PRIVATE;

@Module public class PreferenceServiceModule {
  @Provides static SharedPreferences provideSharedPreferences(Application context) {
    return context.getSharedPreferences("Red", MODE_PRIVATE);
  }
}

File: app/src/main/java/com/benoitquenaudon/tvfoot/red/injection/module/PreferenceServiceModule.kt
package com.benoitquenaudon.tvfoot.red.injection.module

import android.content.Context.MODE_PRIVATE
import android.content.SharedPreferences
import com.benoitquenaudon.tvfoot.red.RedApp
import dagger.Module
import dagger.Provides

@Module
object PreferenceServiceModule {

  @JvmStatic
  @Provides
  fun provideSharedPreferences(context: RedApp): SharedPreferences {
    return context.getSharedPreferences("Red", MODE_PRIVATE)
  }
}
